routing/kbucket/table.go

```diff
@@ -155,9 +155,10 @@ func (rt *RoutingTable) NearestPeers(id ID, count int) []peer.ID {
 	bucket = rt.Buckets[cpl]
 
 	var peerArr peerSorterArr
-	if bucket.Len() == 0 {
-		// In the case of an unusual split, one bucket may be empty.
-		// if this happens, search both surrounding buckets for nearest peer
+	peerArr = copyPeersFromList(id, peerArr, bucket.list)
+	if len(peerArr) < count {
+		// In the case of an unusual split, one bucket may be short or empty.
+		// if this happens, search both surrounding buckets for nearby peers
```

Comment:
this will end up with "more than `count`" peers, as it will copy from both other buckets, and copy doesn't have a limit.